api/v1beta2/ssp_types.go

```diff
@@ -104,11 +104,14 @@ type TektonTasks struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
-// FeatureGates defines feature gate for tto operator
+// FeatureGates for SSP
 type FeatureGates struct {
 	DeployTektonTaskResources bool `json:"deployTektonTaskResources,omitempty"`
 
 	DeployVmConsoleProxy bool `json:"deployVmConsoleProxy,omitempty"`
+
+	// Enables deployment of the common-instancetypes bundles, defaults to true.
+	DeployCommonInstancetypes *bool `json:"deployCommonInstancetypes,omitempty"`
```

Comment:
bools empty/default value is `false` so this needs to be a pointer with an empty value so we can determine when an operator has actually requested that the feature gate be disabled. IMHO all of these should be bool pointers.

Reply:
ACK

Do we need a follow up to change the others? Do other features gates cleanup too, if set to false?

Reply:
Yes, `DeployTektonTaskResources` doesn't cleanup [1] while `DeployVmConsoleProxy` does [2], I had a TODO to refactor the `CleanupResult` to `ReconcileResult` translation as well from the latter FWIW.

Reply:
ACK

Reply:
Btw. tekton tasks does not clean up objects intentionally, when feature gate is disabled. It is because user could already create pipelineRun, which might reference our objects.

Reply:
The default value can be specified in CRD, by adding a kubebuilder annotation in the code:
```
//+kubebuilder:default=true
DeployCommonInstancetypes *bool `json:"deployCommonInstancetypes,omitempty"`
```

Reply:
Ok, I see. Then you can ignore my comment.